tests/e2e/e2e_utils.py

```
@@ -39,33 +34,196 @@ def run_oneshot_for_e2e_testing(
+            "torch.distributed.run",
+            "--nproc_per_node",
+            str(num_gpus),
+            "--log-dir",
+            "/tmp/torchrun-logs",
+            "--tee",
+            "3",
+            _RUNNER,
+            json.dumps(config),
+        ]
+        logger.info(f"========== RUNNING DDP oneshot ({num_gpus} GPUs) ==========")
+        result = subprocess.run(cmd, capture_output=True, text=True, env=os.environ.copy())
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)